File: javastory.budgetsh.stage4/src/javastory/budgetsh/stage4/server/budget/store/CashBookStore.java
package javastory.budgetsh.stage4.server.budget.store;

import java.util.ArrayList;

import javastory.budgetsh.stage4.server.budget.entity.budget.CashBook;

public interface CashBookStore {
	//
	public boolean exist(String bankAccount);
	public CashBook regist(CashBook cashbook);
	public CashBook retrieve(String account);
	public void update(CashBook foundCashbook);
	public void remove(CashBook foundCashbook);
	public ArrayList<CashBook> retrieveAll();

}

File: javastory.budgetsh.stage5.1/src/test/CashBookStore.js
const MapStorage = require('./MapStorage.js');
const mapStorage = new MapStorage();

class CashBookStore{
	//
	constructor(){
		
	}
	exist(bankAccount){
		if(mapStorage.cashbookMap.has(bankAccount)){
			return true;
		}else{
			return false;
		}
	}
	
	retrieve(bankAccount){
		return mapStorage.cashbookMap.get(bankAccount);
	}
	
	update(cashbook){
		mapStorage.cashbookMap.delete(cashbook.bankAccount);
		mapStorage.cashbookMap.set(cashbook.bankAccount, cashbook);
	}
	
	remove(cashbook){
		mapStorage.cashbookMap.delete(cashbook.bankAccount);
	}
	
	regist(cashbook){
		mapStorage.cashbookMap.set(cashbook.bankAccount, cashbook);
	}
	
}

module.exports = CashBookStore;